### PR TITLE
NEW + DOC: image inversion

### PR DESCRIPTION
**Pull request by sciunto**
Looks like this point has to be documented somewhere.

https://stackoverflow.com/questions/28084908/python-with-scikit-image-to-invert-colors-back-and-white

### CODE REVIEW

File: doc/source/user_guide/transforming_image_data.rst [hunk 1]
```diff
@@ -81,12 +81,24 @@ from RGB to grayscale::
...
+~~~~~~~~~~~~~~~
+
+An inverted image is also called complementary image. For binary images, True values
+become False and conversely. For grayscale images, pixel values are replaced by the
+difference of the maximum value of the data type and the actual value. For RGB
+images, the same operation is done for each channel. This operation can be achieved
+with the negation operator ~ on any numpy array::
+
+    >>> img = data.camera()
+    >>> inverted_img = ~img
```
Comment (soupault) [inline doc/source/user_guide/transforming_image_data.rst]: This doesn't work for `float` dtypes. Maybe be more precise here?

Comment (jni) [inline doc/source/user_guide/transforming_image_data.rst]: Yes, and I would say that this is *not* the desired operation anyway. The inverted image should be the *dtype* maximum, minus the image.

**Comment by sciunto** [comment]
I pushed a new function `skimage.util.invert`. Very simple but probably also very useful...

**Comment by jni** [comment]
@sciunto LGTM, nice and clean implementation, but can you clean up the commit history? =)

**Comment by sciunto** [comment]
Fixed :) thanks Don't know what happened.

**Comment by jni** [comment]
Ok unqualified LGTM now! =)

File: skimage/util/tests/test_invert.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import numpy as np
+from numpy.testing import assert_array_equal
+from skimage import dtype_limits
+from skimage.util.invert import invert
```
Comment (soupault) [inline skimage/util/tests/test_invert.py]: I think it is better to rename `invert.py` to `_invert.py` and use `from skimage.util import invert` here.

Comment (JDWarner) [inline skimage/util/tests/test_invert.py]: :+1:

**Comment by soupault** [comment]
LGTM now.

**Pull request merged**